### DISCUSSION TITLE
how to use this library with a real-time data stream?

### DISCUSSION THREAD

**Discussion by tiger2014** [DaveSkender/Stock.Indicators#813] [Ideas]
Hi Dave,

I'm using your library in a real-time stock data stream,  I have to re-calculate the total series when every new candle comes in.

Is there a method that doesn't need to re-calculate all the data series, but only calculates the new coming one?

my code is :

`While(true)`
`{`
    `.....`
    `dataList.Add(newCandles);`
    `dataList.RemoveAt(0);`
    `dataList.GetMacd(9, 24, 5, CandlePart.Close);`
`}`

the count of the dataList is keeping in 300.

Thanks!

**Comment by DaveSkender** [comment]
See our [discussion on incremental calculations](https://github.com/DaveSkender/Stock.Indicators/discussions/216#discussioncomment-2444760).  I have streaming use cases in our backlog and will have a preview of how this would work in #802.  Currently, the library only calculates full series.  The interface will be something like:

```csharp
# establish indicator base
var emaBase = quotes.InitEma(14);
var results = emaBase.Results;

# add streaming / repeating updates
emaBase.Add(quote);
```

**Reply by tiger2014** [reply]
Thanks

**Comment by DaveSkender** [comment]
Follow or continue this conversation in our [discussion on streaming indicators](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) for progress and information about pre-release package versions to support live and incremental quote sources.
